lib/user.cpp

```diff
@@ -424,3 +427,15 @@ void User::processEvent(const RoomMemberEvent& event, const Room* room)
         updateAvatarUrl(event.avatarUrl(), d->avatarUrlForRoom(room), room);
     }
 }
+
+qreal User::hueF() const
+{
+    QByteArray hash = QCryptographicHash::hash(d->userId.toUtf8(),
+                                               QCryptographicHash::Sha1);
+    QDataStream dataStream(qToLittleEndian(hash).left(2));
+    dataStream.setByteOrder(QDataStream::LittleEndian);
+    quint16 hashValue;
+    dataStream >> hashValue;
+    qreal hueF = static_cast<qreal>(hashValue)/std::numeric_limits<quint16>::max();
```

Comment:
I would additionally discretize `hueF` down to a limited number of values, to avoid an uneasy feeling when two (same or different, doesn't matter) names have _almost_ the same name. Say, down to 256 values :)

Reply:
From my point of view, even quint16 resolution level is too rough for modern displays and even smaller color difference is handy :) But I agree to follow this proposal, test it in real environment and provide additional changes if case of anything.
I've also found (thanks to that discussion) that my convertion to qreal is probably wrong: I should also multiply to qreal max value to save the proportion. Fixed it too.

Reply:
My mistake at previous message: I should not additionally multiply to qreal max value since my idea was to follow [QColor logic](https://doc.qt.io/qt-5/qcolor.html#integer-vs-floating-point-precision):
```
0 <= hue <= 359
0 <= hueF <= 1 (QML friendly)
```
Fixed and noted it at method description.

Summarizing, for now I've moved logic to `User::Private`, added both `hue` and `hueF` properties and documented it, while switched back to `quint16` resolution following xep-0392 and the info that QColor components are stored using 16-bit integers.

Still could be wrong at private part implementation and still ready to simplify gradation to 256 values (1 byte from hash instead of 2 currently)

Reply:
Good! For the sake of self-checking, I recommend adding `Q_ASSERT`s before returning the value in `makeHueF()`, to be sure that it at least doesn't fall out of `0 <= hueF <= 1` range. Running a client through a crowded enough account will then immediately tell you whether the invariants stand.
...Hm, by the way, try to make a thought experiment: assume the 2 bytes from the hash are `0xFFFF` - what hue (integer) value do they correspond to, according to your code?

Reply:
my bad, added `Q_ASSERT` and changed multiplier to `359`